spec/classes/trusted_external_command_spec.rb

```diff
@@ -23,14 +23,7 @@
     it { is_expected.to compile }
   end
 
-  # This test only works with the !defined() syntax for checking if a resource
-  # exists. Unfortunately, that function does not work properly when actually run
-  # against a master. When run in a real environment the syntax needs to be
-  # unless Service['pe-puppetserver'] {blah}. But that syntax doesn't work for
-  # this test. Until a way is found to fix this test I would prefer the syntax
-  # that works in the product be used until we can either fix this test or find
-  # a manifest syntax that will work in both cases.
-  # context "when Service['pe-puppetserver'] is undefined" do
-  #   it { is_expected.to compile }
-  # end
+  context "when Service['pe-puppetserver'] is undefined" do
```

Comment:
I use `puppet apply` when applying the `trusted_external_command` class so I think this test is useful to make sure that use-case works.

Reply:
It's not that the test isn't useful, it's that it doesn't work properly if a puppet run is called via `puppet agent -t`. Using puppet apply will work fine to verify that the class can configure a master properly, but using puppet apply does not trigger the external command to actually be called. To trigger to external command to be called and return data so that things like `$trusted` data are populated, an agent run must be triggered via puppet agent -t. Then what you find is that if you use `defined` as the guard for `Service['pe-puppetserver]` resource the run fails because it doesn't properly protect against duplicate resource declarations like you would hope it would. So, to get a real puppet agent -t run to work you have to remove `defined` and use something else that really will guard against duplicate resource declarations. Unfortunately I haven't yet found a syntax to use in the manifest that will both fix puppet runs and work with this test.

Reply:
Based on the feedback from @reidmv aboce this test would be removed anyway because the assumption is that this resource would always be declared by some other file.

Reply:
Yup, just saw his feedback. I'll remove the test.